### PR TITLE
Impr/multipleimagesource

### PR DESCRIPTION
**Pull request by emelalkim**

### CODE REVIEW

File: plugins/DICOMwebServer.js [hunk 1]
```diff
@@ -906,12 +917,15 @@ async function dicomwebserver(fastify) {
     'getStudySeriesDIMSE',
     (studyUID) =>
       new Promise((resolve, reject) => {
+        console.time('dimse');
```
Comment (emelalkim) [inline plugins/DICOMwebServer.js:920]: clean up

File: plugins/DICOMwebServer.js [hunk 2]
```diff
@@ -906,12 +917,15 @@ async function dicomwebserver(fastify) {
...
+        console.time('dimsecall');
         const dimsePromises = [
           fastify.promisifyDIMSE(config.dimse, studyUID),
-          fastify.promisifyDIMSE(config.vnaDimse, studyUID),
+          fastify.promisifyDIMSE(config.archiveDimse, studyUID),
         ];
         Promise.all(dimsePromises).then((results) => {
           try {
+            console.timeEnd('dimsecall');
```
Comment (emelalkim) [inline plugins/DICOMwebServer.js:928]: clean up

File: plugins/DICOMwebServer.js [hunk 3]
```diff
@@ -921,19 +935,35 @@ async function dicomwebserver(fastify) {
...
+              }, {});
+              // fill in the series descriptions retrieved from Sectra
+              res = res.map((item) => {
+                if (item['0020000E'] && item['0020000E'].Value && item['0020000E'].Value[0])
+                  // eslint-disable-next-line no-param-reassign
+                  item['0008103E'] = map[item['0020000E'].Value[0]];
+                return item;
+              });
             }
+            console.timeEnd('dimse');
```
Comment (emelalkim) [inline plugins/DICOMwebServer.js:966]: clean up

File: plugins/DICOMwebServer.js [hunk 4]
```diff
@@ -1074,20 +1104,49 @@ async function dicomwebserver(fastify) {
...
-          this.request
-            .get(
-              `${config.dicomWebConfig.qidoSubPath}/studies/${params.study}/series/${params.series}/instances?includefield=00280008`,
-              header
+          // Get sectra, then vna with qidoSubPath
+          fastify
+            .queryQIDO(
+              `${config.dicomWebConfig.qidoSubPath}/studies/${params.study}/series/${params.series}/instances?includefield=00280008`
             )
             .then(async (response) => {
```
Comment (emelalkim) [inline plugins/DICOMwebServer.js:1146]: change response to res to avoid confusion

**Review by OzgeYurtsever — APPROVED** [review]

**Pull request merged**